registry/registry-local/src/test/java/com/alipay/sofa/rpc/registry/local/DomainRegistryTest.java

```diff
@@ -0,0 +1,316 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.alipay.sofa.rpc.registry.local;
+
+import com.alipay.sofa.rpc.client.ProviderGroup;
+import com.alipay.sofa.rpc.client.ProviderHelper;
+import com.alipay.sofa.rpc.client.ProviderInfo;
+import com.alipay.sofa.rpc.common.RpcConstants;
+import com.alipay.sofa.rpc.config.ConsumerConfig;
+import com.alipay.sofa.rpc.config.ProviderConfig;
+import com.alipay.sofa.rpc.config.RegistryConfig;
+import com.alipay.sofa.rpc.listener.ProviderInfoListener;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.CountDownLatch;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNotEquals;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertSame;
+import static org.junit.Assert.assertTrue;
+
+/**
+ * @author <a href=mailto:orezsilence@163.com>zhangchengxi</a>
+ */
+public class DomainRegistryTest {
+
+    private DomainRegistry domainRegistry;
+
+    @Before
+    public void before() {
+        domainRegistry = new DomainRegistry(new RegistryConfig());
+    }
+
+    @After
+    public void after() {
+        domainRegistry.destroy();
+    }
+
+    @Test
+    public void testInit() {
+        DomainRegistry domainRegistry = new DomainRegistry(new RegistryConfig());
+        assertNull(domainRegistry.scheduledExecutorService);
+        domainRegistry.init();
+        assertTrue(domainRegistry.scheduledExecutorService.isStarted());
+    }
+
+    @Test
+    public void testDirectUrl2IpUrl() {
+        ProviderInfo providerInfo = ProviderHelper.toProviderInfo("bolt://alipay.com:12200");
+        List<ProviderInfo> providerInfos = domainRegistry.directUrl2IpUrl(providerInfo, null);
+        assertTrue(providerInfos.size() > 0);
+        String host = providerInfos.get(0).getHost();
+        assertNotEquals("alipay.net", host);
```

Comment:
Misspelling，there should be "alipay.com".

Reply:
fixed